File: cplusplus/src/solutions/SolutionFactory.cpp
#include "SolutionFactory.hpp"

// Solutions:
#include "Solution0002.hpp"
#include "Solution0003.hpp"
#include "Solution0005.hpp"
#include "Solution0006.hpp"
#include "Solution0007.hpp"
#include "Solution0008.hpp"
#include "Solution0149.hpp"
#include "Solution1535.hpp"
#include "Solution2956.hpp"
#include "Solution2961.hpp"
#include "SolutionDefault.hpp"

SolutionFactory::SolutionFactory()
{
}

std::shared_ptr<ISolution> SolutionFactory::createSolution(const uint32_t taskNumber)
{
    std::shared_ptr<ISolution> solution{nullptr};
    switch (taskNumber) {
    case 2:
        solution = std::make_shared<Solution0002>();
        break;
    case 3:
        solution = std::make_shared<Solution0003>();
        break;
    case 5:
        solution = std::make_shared<Solution0005>();
        break;
    case 6:
        solution = std::make_shared<Solution0006>();
        break;
    case 7:
        solution = std::make_shared<Solution0007>();
        break;
    case 8:
        solution = std::make_shared<Solution0008>();
        break;
    case 149:
        solution = std::make_shared<Solution0149>();
        break;
    case 1535:
        solution = std::make_shared<Solution1535>();
        break;
    case 2956:
        solution = std::make_shared<Solution2956>();
        break;
    case 2961:
        solution = std::make_shared<Solution2961>();
        break;
    default:
        solution = std::make_shared<SolutionDefault>();
        break;
    }

    solution->link(shared_from_this());
    return solution;
}

File: python/solutions/SolutionFactory.py
from solutions.ISolution import ISolution
from solutions.Solution0002 import Solution0002
from solutions.Solution0003 import Solution0003
from solutions.Solution0005 import Solution0005
from solutions.Solution0006 import Solution0006

class SolutionFactory:
    def createSolution(self, solutionNumber: int) -> ISolution:
        returnSolution = None

        if solutionNumber==2 :
            returnSolution = Solution0002()
        elif solutionNumber == 3 :
            returnSolution = Solution0003()
        elif solutionNumber == 5 :
            returnSolution = Solution0005()
        elif solutionNumber == 6 :
            returnSolution = Solution0006()
        else:
            returnSolution = ISolution()

        return returnSolution